src/module/codegen/codegen

```diff
@@ -45,20 +50,35 @@ SYSCALLS.each do |call, spec|
   # 3. syscalls.gen.x, to set up the initial wrapper
   # 4. syscalls/internal.gen.h, to prototype the internal wrapper
   # 5. syscalls/<syscall>.gen.c, to set up the actual faulty calls
-
+  
   nr = spec["nr"] || call
+  number = "__NR_#{nr}"
+  if PLATFORM == 'freebsd'
+    number = "SYS_#{nr}"
+  end
+  
+  hai "#{call} (nr: #{number})"
+  if PLATFORM == 'freebsd'
+    gen_files[:krf_x].puts <<~KRF_X
+      krf_faultable_table[#{number}].sy_call = (sy_call_t *)&krf_sys_#{call};
+    KRF_X
+  else       
+    gen_files[:krf_x].puts <<~KRF_X
+      krf_faultable_table[#{number}] = (void *)&krf_sys_#{call};
+    KRF_X
+  end
 
-  hai "#{call} (nr: __NR_#{nr})"
-
-  gen_files[:krf_x].puts <<~KRF_X
-    krf_faultable_table[__NR_#{nr}] = (void *)&krf_sys_#{call};
-  KRF_X
-
+  sYSCALL_INSERT = "(void *)krf_sys_call_table[#{number}]"
```

Comment:
`sYSCALL_INSERT` -> `SYSCALL_INSERT`?

Reply:
The all caps means it should be a constant, but it isn't since it is changed on every syscall. I will update it to lowercase